File: 0268-missing-number/0268-missing-number.java
class Solution {
    public int missingNumber(int[] nums) {
        // first can sort using cyclic sort approach

        int i=0;
        int n=nums.length;

        while(i<nums.length){
            if(nums[i]!=i && nums[i]<n){
                int temp=nums[i];
                nums[i]=nums[temp];
                nums[temp]=temp;
            }else{
                i++;
            }
        }

        for(int j=0;j<n;j++){
            if(nums[j]!=j) return j;
        }

        return n;

    }
}

File: 0268-missing-number/0268-missing-number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function(nums) {
    let i = 0;
    let n = nums.length;

    while (i < n) {
        if (nums[i] < n && nums[i] !== nums[nums[i]]) {
            let correctIndex = nums[i];
            [nums[i], nums[correctIndex]] = [nums[correctIndex], nums[i]];
        } else {
            i++;
        }
    }

    for (let i = 0; i < n; i++) {
        if (nums[i] !== i) return i;
    }

    return n;
};